### PR TITLE
numpy deprecation warning filter for v1.24.0

### PR DESCRIPTION
**Pull request by bmorris3**
Numpy v1.24.0 triggers several deprecation warnings in scikit-image. See progress on fixes in scikit-image/scikit-image#6633, and failure in #1834.

This PR tries to filter out that specific deprecation warning.

### Description

This pull request is to address ...

Fixes #<Issue Number>

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by pllim — CHANGES_REQUESTED** [review]
Blanket ignore seems overkill for something that only crops up on dev job.

Lemme catch up on the logs and see if we can do something more localized.

**Comment by bmorris3** [comment]
@pllim – to help you up to speed, see [this warning](https://github.com/spacetelescope/jdaviz/actions/runs/3550655505/jobs/5964421579#step:5:274), for example.

**Comment by pllim** [comment]
You might need to fix the regex.

**Comment by bmorris3** [comment]
@pllim That's what I was worried about, but couldn't figure out how to test it locally. Trying again.

File: setup.cfg [hunk 1]
```diff
@@ -97,7 +97,7 @@ filterwarnings =
     ignore:distutils Version classes are deprecated:DeprecationWarning
     ignore:Passing unrecognized arguments to super:DeprecationWarning
     ignore:.*With traitlets 4\.1, metadata should be set using the \.tag\(\) method:DeprecationWarning
-    ignore:`np.bool8` is a deprecated alias for `np.bool_`.  (Deprecated NumPy 1.24):DeprecationWarning
+    ignore:is a deprecated alias for:DeprecationWarning
```
Comment (pllim) [inline setup.cfg:100]: Maybe a little more specific? 😬 
```suggestion
    ignore:.*np\.bool8.*is a deprecated alias for:DeprecationWarning
```

File: setup.cfg [hunk 2]
```diff
@@ -97,6 +97,8 @@ filterwarnings =
     ignore:distutils Version classes are deprecated:DeprecationWarning
     ignore:Passing unrecognized arguments to super:DeprecationWarning
     ignore:.*With traitlets 4\.1, metadata should be set using the \.tag\(\) method:DeprecationWarning
+    ignore:.*np\.bool8.*is a deprecated alias for:DeprecationWarning
+    ignore:.*np\.unit0.*is a deprecated alias for:DeprecationWarning
```
Comment (pllim) [inline setup.cfg:101]: What is unit0 ? 
```suggestion
    ignore:.*np\.int0.*is a deprecated alias for:DeprecationWarning
```

Comment (bmorris3) [reply setup.cfg]: 🤦🏻

**Comment by pllim** [comment]
Maybe it was `int0` and not `uint0` or both?

`DeprecationWarning: `np.int0` is a deprecated alias for `np.intp`.`

**Comment by pllim** [comment]
Also these are strictly upstream right? Do you plan to fix `uint0` and `int0` upstream too?

File: jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py [hunk 3]
```diff
@@ -17,8 +17,8 @@
 def jwst_header_to_skyregion(header):
     s_region = header['S_REGION']
     footprint = s_region.split("POLYGON ICRS")[1].split()
-    ra = np.array(footprint[::2], dtype=np.float)
-    dec = np.array(footprint[1::2], dtype=np.float)
+    ra = np.array(footprint[::2], dtype=np.float_)
+    dec = np.array(footprint[1::2], dtype=np.float_)
```
Comment (pllim) [inline jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py:21]: Isn't this what numpy actually suggested?

```suggestion
    ra = np.array(footprint[::2], dtype=float)
    dec = np.array(footprint[1::2], dtype=float)
```

**Comment by pllim** [comment]
Ignore the release job failure. It is https://github.com/pypa/twine/issues/940

**Comment by pllim** [comment]
Remaining failures in the devdeps job all use https://stsci.box.com/shared/static/exnkul627fcuhy5akf2gswytud5tazmw.fits . Curious that this file worked with the other remote data job but not this one, but investigating it is out of scope here.

**Review by pllim — APPROVED** [review]

File: jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py [hunk 4]
```diff
@@ -17,8 +17,8 @@
 def jwst_header_to_skyregion(header):
     s_region = header['S_REGION']
     footprint = s_region.split("POLYGON ICRS")[1].split()
-    ra = np.array(footprint[::2], dtype=np.float)
-    dec = np.array(footprint[1::2], dtype=np.float)
+    ra = np.array(footprint[::2], dtype=float)
+    dec = np.array(footprint[1::2], dtype=float)
```
Comment (pllim) [inline jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py:21]: I backported this to v3.1.x in d7ec0ab70c885246be00d8312e7b596b88e286dc

**Pull request merged**